projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts

```diff
@@ -24,4 +26,122 @@ describe(`ComboBox`, () => {
         await input.click();
         await expect(page).toHaveScreenshot(`01-combobox-dont-allow-disabled-04.png`);
     });
+
+    describe(`API`, () => {
+        [true, false].forEach(strict => {
+            test(`search shouldn't be reset if an exact match is entered when strict is ${strict}`, async ({
+                page,
+            }) => {
+                const {apiPageExample} = new TuiDocumentationPagePO(page);
+
+                const {comboboxTextfield} = new TuiComboBoxPO(apiPageExample);
+
+                await visitBy(page, strict);
+                await openFormValue(apiPageExample);
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-${strict}.png`,
+                );
+
+                await comboboxTextfield.click();
+                await page.keyboard.type(`Rubles (500)`);
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-waited-mark-${strict}.png`,
+                );
+
+                await clickFirstOption(page);
+                await focusWrapper(apiPageExample);
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-focused-${strict}.png`,
+                );
+
+                await comboboxTextfield.click();
+                await page.keyboard.press(`Backspace`);
+                await focusWrapper(apiPageExample);
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-backspaced-${strict}.png`,
+                );
+
+                await comboboxTextfield.click();
+                await page.keyboard.press(`Control+A`);
+                await page.keyboard.press(`Backspace`);
+                await focusWrapper(apiPageExample);
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-remove-all-${strict}.png`,
+                );
+            });
+
+            test(`correct word match when strict is ${strict}`, async ({page}) => {
+                const {apiPageExample} = new TuiDocumentationPagePO(page);
+                const {comboboxTextfield} = new TuiComboBoxPO(apiPageExample);
+
+                await visitBy(page, strict);
+                await openFormValue(apiPageExample);
+
+                await comboboxTextfield.click();
+                await page.keyboard.type(`dOlLaRs (237)`);
+                await expect(page).toHaveScreenshot(
+                    `correct-word-match-when-strict-${strict}.png`,
+                );
+
+                await comboboxTextfield.click();
+                await page.keyboard.press(`Backspace`);
+
+                await page.waitForTimeout(200);
+
+                await comboboxTextfield.click();
+                await page.keyboard.press(`Backspace`);
+
+                await page.waitForTimeout(200);
```

Comment:
Don't repeat every code line from previous cypress test 😄 

Is this playwright test actually required this new explicit timeout ?

Drop it and use the following cli-command to locally check if test is flaky without it:

```
nx e2e demo-playwright combobox.spec.ts --repeat-each=50
```